src/Tools/CommandLine.hs

```diff
@@ -0,0 +1,99 @@
+module Tools.CommandLine where
+
+import           System.IO                      ( hPutStrLn
+                                                , stderr
+                                                )
+import           System.Exit                    ( exitFailure )
+import           Data.Version
+import           Options.Applicative
+import           Data.Semigroup                 ( (<>) )
+import qualified Data.Text                     as T
+
+import           Constants.Flags
+import           Paths_madlib                  as Lib
+
+hashBar = "################################################"
+h1 = " ____    __  ____   _____   ____    ____  ______"
+h2 = "|    \\  /  ||    \\  |    \\ |    |  |    ||      )"
+h3 = "|     \\/   ||     \\ |     \\|    |_ |    ||     <"
+h4 = "|__/\\__/|__||__|\\__\\|_____/|______||____||______)"
+headerString = h1 ++ "\n" ++ h2 ++ "\n" ++ h3 ++ "\n" ++ h4
+
+-- -- parseCommand :: Parser Command
+-- -- parseCommand = subparser $ command "transform" parseTransform
+-- -- parseCommand = subparser
+--   -- $ command "transform" (parseTransform `withInfo` "Transform source code")
+
+-- parseOptions :: Parser Options
+-- parseOptions = Options <*> parseInput
+
+-- run :: Options -> IO ()
+-- run (Options command) = do
+--   case command of
+--     Transform i o c l -> (return ())
+--  where
+--   printit :: Show a => (Command -> a) -> Maybe Command -> IO ()
+--   printit acc = maybe (err "fucked up") (print . acc)
+
+-- main :: IO ()
+-- main = run =<< execParser (parseOptions `withInfo` "Cool!")
+
+data TransformFlags = TransformFlags
+  { input      :: FlagInput
+  , output     :: FlagOutput
+  , config     :: FlagConfig
+  -- , stdin      :: Bool
+  -- , literate   :: Literate
+  }
+
+parseConfig :: Parser FlagConfig
+parseConfig = FileConfig <$> strOption
+  (  long "config"
+  <> short 'c'
+  <> metavar "CONFIG"
+  <> help "What config to use"
+  <> showDefault
+  <> value "madlib.yaml"
+  )
+
+fileInput :: Parser FlagInput
+fileInput = FileInput <$> strOption
```

Comment:
That library looks rather cool I think.

Reply:
Yeah, when I first read their README, I was alarmed at how much they were pitching, but it's pretty straightforward once you commit to it.